ci/travis/linux-config.sh

```diff
@@ -24,6 +24,38 @@ fi
 if [[ -n "${IMAGE+x}" ]]; then
   echo "IMAGE is already defined."
 else
-  readonly IMAGE="cached-${DISTRO}-${DISTRO_VERSION}"
-  readonly BUILD_OUTPUT="build-output/${IMAGE}"
+  readonly IMAGE="${DISTRO}-${DISTRO_VERSION}"
+  suffix=""
+  if [[ -n "${CC:-}" ]]; then
+    suffix="${suffix}-${CC}"
+  fi
+  if [[ -n "${BUILD_TYPE+x}" ]]; then
+    suffix="${suffix}-${BUILD_TYPE}"
+  else
+    suffix="${suffix}-Release"
+  fi
+  if [[ "${SCAN_BUILD+x}" = "yes" ]]; then
+    suffix="${suffix}-scan"
+  fi
+  if [[ -z "${CMAKE_FLAGS:-}" ]]; then
+      /bin/true
+  elif echo "${CMAKE_FLAGS}" | grep -Eq BUILD_SHARED_LIBS=yes; then
```

Comment:
Are all these mutually exclusive?

Reply:
No, we could have more than one set, though we rarely (ever?) do.